core/dvector.h

```diff
@@ -264,6 +264,9 @@ class DVector {
 
 	DVector<T> subarray(int p_from, int p_to) {
 
+		if (p_to==-1 && size()>0) {
```

Comment:
Can't you use #4861 instead? Because if we're supporting -1, then we should support -2, -3, etc.